backend/src/main/java/botobo/core/ui/auth/PathMatcherInterceptor.java

```diff
@@ -0,0 +1,47 @@
+package botobo.core.ui.auth;
+
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class PathMatcherInterceptor implements HandlerInterceptor {
+    private final HandlerInterceptor handlerInterceptor;
+    private final PathPatterns pathPatterns;
+
+    public PathMatcherInterceptor(HandlerInterceptor handlerInterceptor) {
+        this.handlerInterceptor = handlerInterceptor;
+        this.pathPatterns = new PathPatterns();
+    }
+
+    public PathMatcherInterceptor addPathPatterns(String pathPattern, PathMethod... pathMethods) {
+        pathPatterns.addPathPatterns(pathPattern, pathMethods);
+        return this;
+    }
+
+    public PathMatcherInterceptor excludePathPatterns(String pathPattern, PathMethod... pathMethods) {
+        pathPatterns.excludePathPatterns(pathPattern, pathMethods);
+        return this;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        String uri = request.getRequestURI();
+        String httpMethod = request.getMethod();
+        if (pathPatterns.isExcludedPath(uri, httpMethod)) {
```

Comment:
이것도 제안 입니다. PathMethod 라는 객체가 존재하니까 활용하는 방향은 어떤가요?
물론 코드가 몇 군데 바꾸긴 해야 돼요🧐 
```suggestion
if (pathPatterns.isExcludedPath(uri, PathMethod.of(httpMethod))) {
        return true;
}
```